src/components/Button.js

```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import Proptypes from "prop-types";
+
+function Button(props) {
+    const { type = "default", children, icon, ...rest } = props;
```

Comment:
The extraction you did in line 5, we can directly extract all props in line 4 itself.
For eg function Button({ type = "default", children, icon, ...rest })